src/content/doc-surrealql/datamodel/idioms.mdx

```diff
@@ -0,0 +1,601 @@
+
+```surql
+SELECT results[WHERE score > 70].score FROM student;
+```
+
+```json title="Response"
+[
+	{
+		results: {
+			score: [
+				76,
+				83,
+				73
+			]
+		}
+	}
+]
+```
+This idiom selects elements in `results.score` where the `score` is greater than 70.
+
+### Method chaining
+
+To call a method on the current data, use a dot `.` followed by the method name and parentheses `()` with arguments. SurrealDB supports method chaining, so you can call multiple methods(functions) on the same data. Learn more about [method chaining](/docs/surrealql/functions/database#method-syntax) in the functions section.
+
+For example, let's create a new `person` record 
+
+```surql title="Create a new person record"
+CREATE person CONTENT {
+    name: "John Doe",
+    age: 30,
+    address: {
+      city: "New York",
+      country: "USA"
+    }
+};
+```
+
+```json title="Response"
+[
+  {
+    "person": {
+      "name": "John Doe",
+      "age": 30,
+      "address": {
+        "city": "New York",
+        "country": "USA"
+      }
+    }
+  }
+]
+```
+
+To call the `uppercase()` method on the `name` field, you can use the following idiom:
+
+```surql
+SELECT name.uppercase() FROM person;
+```
+
+```json title="Response"
+[
+  {
+    "name": "JOHN DOE"
+  }
+]
+```
+
+In the example above, `uppercase()` is a method called on `person.name` to convert it to uppercase. Although this method is called as `.uppercase()` it is actually the [`string::uppercase()`](/docs/surrealql/functions/database/string#stringuppercase) function that is called. 
+
+SurrealDB will automatically convert the idiom part `.uppercase()` to the function `string::uppercase()` when the query is executed.
+
+To learn more about string method chaining in SurrealQL, see the [string functions](/docs/surrealql/functions/database/string#method-chaining) section.
+
+### Graph Navigation
+
+SurrealDB can also be used in the context of graph databases, where data is stored and navigated using graph traversal idioms. The [`RELATE` statement](/docs/surrealql/statements/relate) is used to create relationships between records. This allows you to traverse related records efficiently without needing to pull data from multiple tables and merging that data together using SQL JOINs.
+
+
+For example, let's consider the following data:
+
+```surql title="Create a new planet, city, and explorer records"
+CREATE planet:unknown_planet;
+CREATE city:el_dorado          SET name = "El Dorado";
+CREATE explorer:drake          SET name = "Drake";
+CREATE explorer:local_guide    SET name = "Local Guide";
+
+RELATE explorer:drake->discovered->planet:unknown_planet;
+RELATE explorer:drake->visited->city:el_dorado;
+RELATE explorer:local_guide->assisted->explorer:drake;
+
+```
+
+```surql title="Retrieve all relationships from Drake"
+SELECT 
+    *,
+    ->? AS actions, 
+    <-? AS interactions, 
+    <->? AS connections
+FROM explorer:drake;
```

Comment:
I like this example and think the aliases could be renamed a bit to make the output more immediately comprehensible. Interactions in particular I think should be renamed because `inter` implies something reciprocal and that makes me think of the `<->` operator.

I think for this example these alias names might work best:

```
SELECT 
    *,
    ->? AS actions,
    <-? AS was,
    <->? AS involved_in
FROM explorer:drake;
```

Because then the output reads pretty nice. Actions are discovered and visited, Drake `was` `assisted` reads nice, and then in the middle is `involved_in` which feels reciprocal.

```
[
	{
		actions: [
			discovered:8fuslj8gszt5ydt07ewp,
			visited:dlspn4ki0iwvz2ij7496
		],
		id: explorer:drake,
		involved_in: [
			assisted:zivnejrm6cwe57fdud8d,
			discovered:8fuslj8gszt5ydt07ewp,
			visited:dlspn4ki0iwvz2ij7496
		],
		name: 'Drake',
		was: [
			assisted:zivnejrm6cwe57fdud8d
		]
	}
]
```